node_modules/oae-loodle/lib/internal/dao/users.js

```diff
@@ -0,0 +1,349 @@
+    var options = {
+        host: host,
+        port: port,
+        path: '/api/authenticate',
+        method:'POST',
+        headers: {
+            'Content-type': 'application/json',
+            'Content-length': data.length
+        },
+    };
+
+    var response = '';
+    var req = http.request(options, function (res) {
+
+        res.on('data', function (chunk) {
+            response += chunk;
+        });
+
+        res.on('end', function () {
+            // Parse the response to json
+            response = JSON.parse(response);
+
+            if (response.type === 'error') {
+                log().error("Response : ", response);
+                return callback(response);
+            }
+
+            return callback(null, response.data);
+
+        });
+    });
+
+    // Handle the request error
+    req.on('error', function (err) {
+        log().error("Can't authenticate the user in the loodle application : ", err.message);
+    });
+
+    // Write data to request body and send the request
+    req.write(data);
+    req.end();
+
+};
+
+/**
+ * Private function to save the specified token in db, associated with the specified user
+ * @param user
+ * @param token
+ * @private
+ */
+var _saveToken = function (user, token, callback) {
+
+    var query = 'INSERT INTO "LoodleTokenByUser" ("userId", "token") values (?, ?)';
+    var parameters = [
+        user.id,
+        token
+    ];
+
+    Cassandra.runQuery(query, parameters, callback);
+
+};
+
+exports.isInLoodle = function (ctx, userId, loodleId, token, callback) {
+
+    // Get config values
+    var host = LoodleConfig.getValue(ctx.tenant().alias, 'activity', 'host'),
+        port = LoodleConfig.getValue(ctx.tenant().alias, 'activity', 'port');
+
+	var options = {
+        host: host,
+        port: port,
+        path: '/api/user/getLoodles',
+        method:'GET',
+        headers: {
+            'Content-type': 'application/json',
+            'Authorization': 'Bearer ' + token
+        },
+    };
+
+    var response = '';
+    var req = http.request(options, function (res) {
+
+        res.on('data', function (chunk) {
+            response += chunk;
+        });
+
+        res.on('end', function () {
+
+            // Parse the response to json
+            response = JSON.parse(response);
+
+            if (response.type === 'error') {
+                log().error("Response : ", response);
+                return callback(response);
+            }
+
+            var inLoodle = false;
+            for(var index in response.data) {
+            	if (response.data[index] == loodleId)
+            		inLoodle = true;
+            }
+
+            return callback(null, inLoodle);
+
+        });
+    });
+
+    req.on('error', function (err) {
+        log().error("Can't get the loodles associated with the user : ", err.message);
+    });
+
+    req.end();
+
+};
+
+// A REMPLACER PAR UNE VERSION OU ON A JUSTE BESOIN DE FOURNIR L'ID DE L'UTILISATEUR DANS OAE
```

Comment:
Has this method been replaced with the correct version? If so, take the comment out.